x-pack/plugins/maps/public/routing/routes/list/load_list_and_render.js

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { getMapsSavedObjectLoader } from '../../bootstrap/services/gis_map_saved_object_loader';
+import { getToasts } from '../../../kibana_services';
+import { i18n } from '@kbn/i18n';
+import { MapsListView } from './list';
+import { Redirect } from 'react-router-dom';
+
+export const LoadListAndRender = class extends React.Component {
+  state = {
+    hasSavedMaps: null,
+    savedMapsList: null,
```

Comment:
You are not doing anything with `savedMapsList` other then just converting to boolean `const mapsLoaded = !!savedMapsList;`. Instead of storing `savedMapsList` in state, store `mapsLoaded`.